opm/simulators/linalg/gpuistl/ISTLSolverGPUISTL.hpp

```diff
@@ -0,0 +1,253 @@
+    }
+
+    /// Construct a system solver.
+    /// \param[in] simulator   The opm-models simulator object
+    explicit ISTLSolverGPUISTL(const Simulator& simulator)
+        : ISTLSolverGPUISTL(simulator, FlowLinearSolverParameters(), false)
+    {
+    }
+
+
+
+    void eraseMatrix() override
+    {
+        // Nothing, this is the same as the ISTLSolver
+    }
+
+    void setActiveSolver(int num) override
+    {
+        if (num != 0) {
+            OPM_THROW(std::logic_error, "Only one solver available for the GPU backend.");
+        }
+    }
+
+    int numAvailableSolvers() const override
+    {
+        return 1;
+    }
+
+    void prepare(const SparseMatrixAdapter& M, Vector& b) override
+    {
+        prepare(M.istlMatrix(), b);
+    }
+
+    void prepare(const Matrix& M, Vector& b) override
+    {
+        try {
+            updateMatrix(M);
+            updateRhs(b);
+        }
+        OPM_CATCH_AND_RETHROW_AS_CRITICAL_ERROR("This is likely due to a faulty linear solver JSON specification. "
+                                                "Check for errors related to missing nodes.");
+    }
+
+    void setResidual(Vector&) override
+    {
+        // Should be handled in prepare() instead.
+    }
+
+    void getResidual(Vector& b) const override
+    {
+        if (!m_rhs) {
+            OPM_THROW(std::runtime_error, "m_rhs not initialized, prepare(matrix, rhs); needs to be called");
+        }
+        m_rhs->copyToHost(b);
+    }
+
+    void setMatrix(const SparseMatrixAdapter&) override
+    {
+        // Should be handled in prepare() instead.
+    }
+
+    bool solve(Vector& x) override
+    {
+        // TODO: Write matrix to disk if needed
+        Dune::InverseOperatorResult result;
+        if (!m_matrix) {
+            OPM_THROW(std::runtime_error, "m_matrix not initialized, prepare(matrix, rhs); needs to be called");
+        }
+        if (!m_rhs) {
+            OPM_THROW(std::runtime_error, "m_rhs not initialized, prepare(matrix, rhs); needs to be called");
+        }
+        if (!m_gpuSolver) {
+            OPM_THROW(std::runtime_error, "m_gpuFlexibleSolver not initialized, prepare(matrix, rhs); needs to be called");
+        }
+    
+        if (!m_x) {
+            m_x = std::make_unique<GpuVector<real_type>>(x);
+        } else {
+            m_x->copyFromHost(x);
+        }
+        m_gpuSolver->apply(*m_x, *m_rhs, result);
+    
+        m_x->copyToHost(x);
+
+        ++m_solveCount;
+
+
+        m_lastSeenIterations = result.iterations;
+        return checkConvergence(result);
+    }
+
+    int iterations() const override
+    {
+        return m_lastSeenIterations;
+    }
+
+    const CommunicationType* comm() const override
+    {
+        // TODO: Implement this if needed
+        return nullptr;
+    }
+
+    int getSolveCount() const override
+    {
+        return m_solveCount;
+    }
+
+private:
+    bool checkConvergence(const Dune::InverseOperatorResult& result) const
```

Comment:
This approach, avoiding having a converged_ member is a good improvement, please also make the identical change in the other ISTLSolver variants! Could be done by putting this implementation as a protected helper in the base class.